src/main/scala/com/test/TestHttpApp.scala

```diff
@@ -140,15 +140,10 @@ class TestHttpApp[F[_] <: AnyRef : Sync : ContextShift](blocker: Blocker, contex
 }
 
 object TestHttpApp {
-  class IOApp(blocker: Blocker)(implicit cs: ContextShift[IO]) extends TestHttpApp[IO](blocker, Http4sWebContext.ioInstance)
+  class App[F[_] <: AnyRef : Sync]( dispatcher: Dispatcher[F] )
+    extends TestHttpApp[F](Http4sWebContext.withDispatcherInstance(dispatcher))
 
-  class TaskApp(blocker: Blocker)(implicit s: Scheduler, permit: CanBlock) extends TestHttpApp[Task](
-    blocker,
-    (request, config) => new Http4sWebContext[Task](request, config.getSessionStore, _.runSyncUnsafe())
-  )
-
-  class ZIOApp(blocker: Blocker)(implicit runtime: zio.Runtime[Any]) extends TestHttpApp[zio.Task](
-    blocker,
-    (request, config) => new Http4sWebContext[zio.Task](request, config.getSessionStore, runtime.unsafeRun(_))
```

Comment:
I would still keep the example using ZIO runtime as a native way of using ZIO instead of Dispatcher. For me it looks like a more ZIO way, but I'm not really an expert in the topic.